### ISSUE TITLE
Failure running `code-server --link` after upgrade to 3.10

### ISSUE THREAD

**Issue by pkufaraway** [cdr/code-server#3342]
## OS/Web Information

- Web Browser: Chrome 90.0.4430.212
- Local OS: WIN10
- Remote OS: WSL Ubuntu 20.04 LTS
- Remote Architecture: A Ubuntu WIN10 sub system on same computer
- `code-server --version`: 3.10.0 6610bff3216c48fdd0884191338b1092765dbb5a

## Steps to Reproduce

`code-server --link` just does not work for me, it was working in previous version, I updated to 3.10 this afternoon and it stopped working.

My command to update it:
```
curl -fsSL https://code-server.dev/install.sh | sh -s
```

## Expected

A correctly running server

## Actual

It failed starting the server.

## Logs

<!--
First run code-server with at least debug logging (or trace to be really
thorough) by setting the --log flag or the LOG_LEVEL environment variable. -vvv
and --verbose are aliases for --log trace. For example:

code-server --log debug

Once this is done, replicate the issue you're having then collect logging
information from the following places:

    1. The most recent files from ~/.local/share/code-server/coder-logs.
    2. The browser console.
    3. The browser network tab.

Console output

[2021-05-11T00:36:56.728Z] trace child:2015 got message {"message":{"type":"handshake","args":{"_":[],"bind-addr":"127.0.0.1:8080","auth":"none","password":"**********************","config":"/home/allenliu/.config/code-server/config.yaml","verbose":true,"link":{},"log":"trace","user-data-dir":"/home/allenliu/.local/share/code-server","extensions-dir":"/home/allenliu/.local/share/code-server/extensions","host":"localhost","port":0,"proxy-domain":[],"usingEnvPassword":false,"usingEnvHashedPassword":false}}}
[2021-05-11T00:36:56.731Z] info  code-server 3.10.0 6610bff3216c48fdd0884191338b1092765dbb5a
[2021-05-11T00:36:56.731Z] info  Using user-data-dir ~/.local/share/code-server
[2021-05-11T00:36:56.731Z] trace Using extensions-dir ~/.local/share/code-server/extensions
[2021-05-11T00:36:56.744Z] info  Using config file ~/.config/code-server/config.yaml
[2021-05-11T00:36:56.744Z] info  HTTP server listening on http://127.0.0.1:35161 (randomized by --link)
[2021-05-11T00:36:56.744Z] info    - Authentication is disabled (disabled by --link)
[2021-05-11T00:36:56.744Z] info    - Not serving HTTPS (disabled by --link)
[2021-05-11T00:36:56.744Z] debug running agent with bind,--code-server-addr=127.0.0.1:35161,
[2021-05-11T00:36:56.910Z] info  Failed to register server: Agent version () is incompatible. Version v0.2.1 or later is required. Try updating to the latest release at https://github.com/cdr/code-server/releases
[2021-05-11T00:36:56.913Z] error --link agent exited with 1
[2021-05-11T00:36:56.913Z] debug child:2015 disposing {}

## Screenshot

![image](https://user-images.githubusercontent.com/16312073/117741207-b8f28d80-b1b6-11eb-8b66-b3399da735f2.png)

## Notes

This issue can be reproduced in VS Code: No

**Comment by pkufaraway** [comment]
Logs:
```
[2021-05-11T00:38:58.704Z] trace child:2522 got message {"message":{"type":"handshake","args":{"_":[],"bind-addr":"127.0.0.1:8080","auth":"none","password":"**********************","config":"/home/allenliu/.config/code-server/config.yaml","verbose":true,"link":{},"log":"trace","user-data-dir":"/home/allenliu/.local/share/code-server","extensions-dir":"/home/allenliu/.local/share/code-server/extensions","host":"localhost","port":0,"proxy-domain":[],"usingEnvPassword":false,"usingEnvHashedPassword":false}}}
[2021-05-11T00:38:58.707Z] info  code-server 3.10.0 6610bff3216c48fdd0884191338b1092765dbb5a
[2021-05-11T00:38:58.707Z] info  Using user-data-dir ~/.local/share/code-server
[2021-05-11T00:38:58.707Z] trace Using extensions-dir ~/.local/share/code-server/extensions
[2021-05-11T00:38:58.720Z] info  Using config file ~/.config/code-server/config.yaml
[2021-05-11T00:38:58.720Z] info  HTTP server listening on http://127.0.0.1:36103 (randomized by --link)
[2021-05-11T00:38:58.720Z] info    - Authentication is disabled (disabled by --link)
[2021-05-11T00:38:58.721Z] info    - Not serving HTTPS (disabled by --link)
[2021-05-11T00:38:58.721Z] debug running agent with bind,--code-server-addr=127.0.0.1:36103,
[2021-05-11T00:38:58.846Z] info  Failed to register server: Agent version () is incompatible. Version v0.2.1 or later is required. Try updating to the latest release at https://github.com/cdr/code-server/releases
[2021-05-11T00:38:58.848Z] error --link agent exited with 1
[2021-05-11T00:38:58.849Z] debug child:2522 disposing {}
```

Screenshot:
![image](https://user-images.githubusercontent.com/16312073/117741358-21416f00-b1b7-11eb-9a9e-43ee92e1b5e6.png)

**Comment by mtxweb** [comment]
same problem for me. I had to reinstall 3.9.3

**Comment by yliu224** [comment]
Same issue here

**Comment by oxy** [comment]
This was because the version field was not set on v0.2.2 of cloud-agent.
I've released v0.2.3 to address this - this will be in a 3.10.1 patch release that will be out in the coming days, but until then, a quick fix would be to replace `/usr/lib/code-server/lib/coder-cloud-agent` with the latest release from `https://github.com/cdr/cloud-agent/releases/latest/download/cloud-agent-$OS-$ARCH`. (`$OS` is `darwin` for macOS / `linux` for linux, and `$ARCH` is amd64/arm64)

For example:
`sudo curl -fsSL "https://github.com/cdr/cloud-agent/releases/latest/download/cloud-agent-linux-amd64" -o /usr/lib/code-server/lib/coder-cloud-agent`

**Comment by naseefalikp6273** [comment]
naseefali@NASEEFs-MacBook-Pro ~ % sudo curl -fsSL "https://github.com/cdr/cloud-agent/releases/latest/download/cloud-agent-linux-amd64" -o /usr/lib/code-server/lib/coder-cloud-agent
Password:
curl: (23) Failed writing body (0 != 943)

**Comment by naseefali** [comment]
sudo curl -fsSL "https://github.com/cdr/cloud-agent/releases/latest/download/cloud-agent-linux-amd64" -o /usr/lib/code-server/lib/coder-cloud-agent
Password:
curl: (23) Failed writing body (0 != 943)

**Comment by jsjoeio** [comment]
@naseefali it looks like you're on macOS so that won't work. If you installed code-server with homebrew then you need to run:
`sudo curl -fsSL "https://github.com/cdr/cloud-agent/releases/latest/download/cloud-agent-darwin-amd64" -o /usr/local/Cellar/code-server/3.10.0/libexec/lib/coder-cloud-agent` and type in your password.

**Comment by naseefali** [comment]
> @naseefalikp6273 it looks like you're on macOS so that won't work. If you installed code-server with homebrew then you need to run:
> `sudo curl -fsSL "https://github.com/cdr/cloud-agent/releases/latest/download/cloud-agent-darwin-amd64" -o /usr/local/Cellar/code-server/3.10.0/libexec/lib/coder-cloud-agent` and type in your password.



naseefali@NASEEFs-MacBook-Pro t_login % code-server --link[2021-05-13T17:58:46.253Z] info  code-server 3.10.0 f9b6d04cab0463c487e0390fc73e902e1f83a99e
[2021-05-13T17:58:46.255Z] info  Using user-data-dir ~/.local/share/code-server
[2021-05-13T17:58:46.269Z] info  Using config file ~/.config/code-server/config.yaml
[2021-05-13T17:58:46.269Z] info  HTTP server listening on http://127.0.0.1:62626 (randomized by --link)
[2021-05-13T17:58:46.269Z] info    - Authentication is disabled (disabled by --link)
[2021-05-13T17:58:46.270Z] info    - Not serving HTTPS (disabled by --link)
[2021-05-13T17:58:48.275Z] info  Unfortunately we cannot ensure a good user experience with your connection latency (189ms). Efforts are underway to accommodate users in most areas.
[2021-05-13T17:58:48.278Z] error --link agent exited with 1
naseefali@NASEEFs-MacBook-Pro t_login %

**Comment by naseefali** [comment]
> @naseefalikp6273 it looks like you're on macOS so that won't work. If you installed code-server with homebrew then you need to run:
> `sudo curl -fsSL "https://github.com/cdr/cloud-agent/releases/latest/download/cloud-agent-darwin-amd64" -o /usr/local/Cellar/code-server/3.10.0/libexec/lib/coder-cloud-agent` and type in your password.



naseefali@NASEEFs-MacBook-Pro t_login % sudo curl -fsSL "https://github.com/cdr/cloud-agent/releases/latest/download/cloud-agent-darwin-amd64" -o /usr/local/Cellar/code-server/3.10.0/libexec/lib/coder-cloud-agent
Password:
naseefali@NASEEFs-MacBook-Pro t_login % code-server --link[2021-05-13T17:58:46.253Z] info  code-server 3.10.0 f9b6d04cab0463c487e0390fc73e902e1f83a99e
[2021-05-13T17:58:46.255Z] info  Using user-data-dir ~/.local/share/code-server
[2021-05-13T17:58:46.269Z] info  Using config file ~/.config/code-server/config.yaml
[2021-05-13T17:58:46.269Z] info  HTTP server listening on http://127.0.0.1:62626 (randomized by --link)
[2021-05-13T17:58:46.269Z] info    - Authentication is disabled (disabled by --link)
[2021-05-13T17:58:46.270Z] info    - Not serving HTTPS (disabled by --link)
[2021-05-13T17:58:48.275Z] info  Unfortunately we cannot ensure a good user experience with your connection latency (189ms). Efforts are underway to accommodate users in most areas.
[2021-05-13T17:58:48.278Z] error --link agent exited with 1
naseefali@NASEEFs-MacBook-Pro t_login %

**Comment by jsjoeio** [comment]
> Unfortunately we cannot ensure a good user experience with your connection latency (189ms). Efforts are underway to accommodate users in most areas.

@naseefali this looks like a different error. My guess is because you are in India and we probably don't have a cluster close to you to ensure a fast connection latency.

**Comment by alliter8** [comment]
I am getting the same error on Linux Mint 20.1:

```
sudo curl -fsSL "https://github.com/cdr/cloud-agent/releases/latest/download/cloud-agent-linux-amd64" -o ~/tmp                                     
curl: (23) Failed writing body (0 != 1369)
```

**Comment by code-asher** [comment]
The fixed version has been released in 3.10.1.
